prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -51,4 +58,108 @@ object FHIRBundleHelpers {
 
         if (targetList.isNotEmpty()) provenanceResource.target.addAll(targetList)
     }
+
+    /**
+     * Gets all properties for a [Base] resource recursively and filters only its references
+     *
+     * @return a list of reference identifiers for a [Base] resource
+     *
+     */
+    fun Base.getResourceReferences(): List<String> {
+        return filterReferenceProperties(this.getResourceProperties())
+    }
+
+    /**
+     * Gets all properties for a [Base] resource recursively
+     *
+     * @return a list of all [Property] for a [Base] resource
+     */
+    fun Base.getResourceProperties(): List<Property> {
+        return this.children().stream().flatMap { getChildProperties(it) }.toList()
+    }
+
+    /**
+     * Filters the [properties] by only properties that have a value and are of type [Reference]
+     *
+     * @return a list containing only the references in [properties]
+     */
+    private fun filterReferenceProperties(properties: List<Property>): List<String> {
```

Comment:
Make this function internal and create unit tests for it

Reply:
Added unit tests